File: 4_semestre/Paradigmas-de-Linguagem-de-Programacao/10_outubro/aula_23-10-25/Java-x- C++-x CSharp-GO-Ruby/biblioteca/Biblioteca.java
package biblioteca;

import java.util.ArrayList;
import java.util.List;

class Biblioteca {
    private List<Livro> livros;

    public Biblioteca() {
        livros = new ArrayList<>();
    }

    public void adicionarLivro(String titulo, String autor) {
        livros.add(new Livro(titulo, autor));
    }

    public void listarLivros() {
        for (int i = 0; i < livros.size(); i++) {
            System.out.println((i + 1) + ". " + livros.get(i));
        }
    }

    public void buscarPorTitulo(String titulo) {
        for (Livro livro : livros) {
            if (livro.getTitulo().equalsIgnoreCase(titulo)) {
                System.out.println("Encontrado: " + livro);
                return;
            }
        }
        System.out.println("Livro não encontrado.");
    }

    public void emprestarLivro(int indice) {
        if (indice >= 0 && indice < livros.size()) {
            Livro livro = livros.get(indice);
            if (!livro.isEmprestado()) {
                livro.emprestar();
                System.out.println("Livro emprestado com sucesso.");
            } else {
                System.out.println("Livro já está emprestado.");
            }
        } else {
            System.out.println("Índice inválido.");
        }
    }

    public void devolverLivro(int indice) {
        if (indice >= 0 && indice < livros.size()) {
            Livro livro = livros.get(indice);
            if (livro.isEmprestado()) {
                livro.devolver();
                System.out.println("Livro devolvido com sucesso.");
            } else {
                System.out.println("Livro não está emprestado.");
            }
        } else {
            System.out.println("Índice inválido.");
        }
    }
}

File: 4_semestre/Paradigmas-de-Linguagem-de-Programacao/aulas/10_outubro/aula_23-10-25/Java-x- C++-x CSharp-GO-Ruby/sistema-biblioteca/biblioteca.rb
require_relative "livro"

class Biblioteca
  def initialize
    @livros = []
  end

  def adicionar_livro(titulo, autor)
    @livros << Livro.new(titulo, autor)
  end

  def listar_livros
    if @livros.empty?
      puts "Nenhum livro cadastrado."
    else
      @livros.each_with_index do |livro, i|
        puts "#{i + 1}. #{livro}"
      end
    end
  end

  def buscar_por_titulo(titulo)
    livro = @livros.find { |l| l.titulo.downcase == titulo.downcase }
    if livro
      puts "Encontrado: #{livro}"
    else
      puts "Livro não encontrado."
    end
  end

  def emprestar_livro(indice)
    if indice >= 0 && indice < @livros.size
      livro = @livros[indice]
      if !livro.emprestado
        livro.emprestar
        puts "Livro emprestado com sucesso."
      else
        puts "Livro já está emprestado."
      end
    else
      puts "Índice inválido."
    end
  end

  def devolver_livro(indice)
    if indice >= 0 && indice < @livros.size
      livro = @livros[indice]
      if livro.emprestado
        livro.devolver
        puts "Livro devolvido com sucesso."
      else
        puts "Livro não está emprestado."
      end
    else
      puts "Índice inválido."
    end
  end
end
